kerby-kerb/kerb-client/src/main/java/org/apache/kerby/kerberos/kerb/client/KrbClientBase.java

```diff
@@ -271,15 +271,23 @@ public void storeTicket(TgtTicket tgtTicket,
      */
     public void storeTicket(SgtTicket sgtTicket, File ccacheFile) throws KrbException {
         LOG.info("Storing the sgt to the credential cache file.");
-        if (!ccacheFile.exists()) {
+        boolean isFreshNew = !ccacheFile.exists();
+        if (isFreshNew) {
             createCacheFile(ccacheFile);
         }
         if (ccacheFile.exists() && ccacheFile.canWrite()) {
-            CredentialCache cCache = new CredentialCache();
+            
             try {
-                cCache.load(ccacheFile);
-                cCache.addCredential(new Credential(sgtTicket, sgtTicket.getClientPrincipal()));
-                cCache.setPrimaryPrincipal(sgtTicket.getClientPrincipal());
+                CredentialCache cCache;
+                if(!isFreshNew){
```

Comment:
Please put spaces before "(" and after ")".

Reply:
I'll fix

Reply:
I should also improve the condition: isFreshNew = !ccacheFile.exists() || (ccacheFile.length() == 0); in order to manage corner cases (opening a file have caused me problems in the past about automatic file creation... of empty files, don't know if this is OS related).